### PR TITLE
Rr4 spinner

### PR DESCRIPTION
**Pull request by Dattaya**
Please, do not yet merge.

### CODE REVIEW

File: docs/js/routing.md [hunk 1]
```diff
@@ -10,16 +10,16 @@ for more information)
 
 ## Usage
 
-To add a new route, use the generator with `npm run generate route`.
-
-This is what a standard (generated) route looks like for a container:
+To add a new route, simply import the `Route` component and use it standalone or inside the `Switch` component (all part of RR4 API):
```
Comment (anuraaga) [inline docs/js/routing.md]: Good place for a link to react-router-v4 docs

File: internals/generators/container/index.js [hunk 2]
```diff
@@ -44,6 +44,11 @@ module.exports = {
     name: 'wantMessages',
     default: true,
     message: 'Do you want i18n messages (i.e. will this component use text)?',
+  }, {
+    type: 'confirm',
+    name: 'wantLoadable',
+    default: true,
+    message: 'Do you want an async loader?',
```
Comment (anuraaga) [inline internals/generators/container/index.js]: Perhaps we should add something like `If you have enabled reducer or sagas, you most likely do.`?

**Review by anuraaga — COMMENTED** [review]
Thanks a lot! Let me know when you're ready to merge, or alternatively I can close my PR and you can open another one - almost none of my code is here anymore since react-loadable is great :)

**Comment by justingreenberg** [comment]
@anuraaga @Dattaya let me know when this lands so I can review main PR... looks solid!

File: docs/js/routing.md (hunk 1 above)
Comment (Dattaya) [reply docs/js/routing.md]: done

File: internals/generators/container/index.js (hunk 2 above)
Comment (Dattaya) [reply internals/generators/container/index.js]: Thanks for reminding me that loader is necessary if we want reducers/sagas, very helpful! I completely redone generators, now `Loadable.js` is always added if a dev wants reducers or sagas.

**Comment by Dattaya** [comment]
Great to hear from you, @justingreenberg, now it should be ready.
@anuraaga, please merge this one.

File: app/routing/Loadable/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      }
+      return Promise.resolve(null);
+    };
+
+    emptyLoadingComponent = () => null;
+
+    loadableComponent = ReactLoadable({
+      ...rest,
+      loader: this.loaderWithAsyncInjectors,
+      LoadingComponent: CustomLoadingComponent || this.context.defaultLoadingComponent || this.emptyLoadingComponent,
```
Comment (anuraaga) [inline app/routing/Loadable/index.js:27]: By the way, I tried running npm-check-updates on this branch and noticed react-loadable 4 is out, and this parameter was renamed to `loading`. Should we use the latest version?

**Review by anuraaga — APPROVED** [review]
Going to merge, thanks a lot! If we want to raise the react-loadable version we can do it later (unless you had a specific reason not to do so).

File: app/routing/Loadable/index.js (hunk 3 above)
Comment (Dattaya) [reply app/routing/Loadable/index.js:27]: also done

**Pull request merged**